File: IT/2/Multithreading/Esercizi/Python/esercizio4.py
# file esercizio4.py

from __future__ import print_function
from multiprocessing import Process
from random import randint

def func():
	print("Ciao, mondo!")

if __name__ == "__main__":
	nProc = randint(2, 10)

	print("Verranno creati {} processi.".format(nProc))

	procs = [Process(target=func) for i in range(nProc)]

	for p in procs:
		p.start()

	for p in procs:
		p.join()

File: IT/2/Multithreading/Esercizi/Java/Esercizio4.java
/* file Esercizio4.java */

import java.util.Random;

class Target implements Runnable {
	@Override
	public void run() {
		System.out.println("Ciao, mondo!");
	}
}

public class Esercizio4 {
	public static void main(String [] args) {
		Random randomizer = new Random(System.currentTimeMillis());
		// tra 2 e 10 thread
		int nThreads = randomizer.nextInt(9) + 2;
		System.out.println("Verranno creati " + nThreads + " thread.");

		for (int i = 0; i < nThreads; i++) {
			Thread t = new Thread(new Target());
			t.start();
		}
	}
}

